components/cloud/gcp/src/gcs.rs

```diff
@@ -134,6 +138,26 @@ impl<T, E: Display> ResultExt for Result<T, E> {
     }
 }
 
+impl DeletableStorage for GcsStorage {
+    fn delete(&self, name: &str) -> LocalBoxFuture<'_, io::Result<()>> {
+        let name = name.to_owned();
+        async move {
```

Comment:
shall we retry here? just like s3Storage. 
BTW should iter_prefix also need retry?